mongo/client.go

```diff
@@ -509,6 +511,23 @@ func (c *Client) configure(opts *options.ClientOptions) error {
 			topology.WithTopologyServerMonitor(func(*event.ServerMonitor) *event.ServerMonitor { return opts.ServerMonitor }),
 		)
 	}
+	// Logger
+	if opts.Logger == nil {
+		var err error
+		opts.Logger, err = mongologger.NewMongoLoggerFromEnv()
```

Comment:
This is a little weird because the rest of the `configure()` code doesn't modify the `opts` parameter. If we want to actually modify `opts`, that should be done in `MergeClientOptions`. Otherwise, this should be

```
logger := opts.Logger
var err error

if logger == nil {
    logger, err = mongologger.NewMongoLoggerFromEnv()
}
if err != nil {
    return err
} 
```